timeseries/src/autogluon/timeseries/models/statsmodels/models.py

```diff
@@ -220,3 +221,147 @@ def _predict_with_local_model(
+                column_index = 0
+            else:
+                coverage = 2 * (1 - q)
+                column_index = 1
+            quantile_pred = sm_model.prediction_intervals(steps=self.prediction_length, alpha=coverage)
+            # Select lower bound of the confidence interval if q < 0.5, upper bound otherwise
+            results.append(quantile_pred.iloc[:, column_index].rename(str(q)))
+        return pd.concat(results, axis=1)
+
+
+class SeasonalNaiveModel(AbstractStatsmodelsModel):
+    """Predicted value is set to the last observed value from the same season (e.g., the same month of the previous year).
+
+    Does not produce a quantile forecast (i.e., all quantiles are set equal to the mean).
+
+
+    Other Parameters
+    ----------------
+    seasonal_period : int or None, default = None
+        Number of time steps in a complete seasonal cycle for seasonal models. For example, 7 for daily data with a
+        weekly cycle or 12 for monthly data with an annual cycle.
+        When set to None, seasonal_period will be inferred from the frequency of the training data. Can also be
+        specified manually by providing an integer > 1.
+        If length of the training series is less than seasonal_period, will predict the mean value.
+    """
+
+    statsmodels_allowed_init_args = [
+        "seasonal_period",
+    ]
+
+    def _update_sm_model_init_args(self, sm_model_init_args: dict, data: TimeSeriesDataFrame) -> dict:
+        sm_model_init_args = sm_model_init_args.copy()
+        seasonal_period = sm_model_init_args.get("seasonal_period", None)
+        if seasonal_period is None:
+            sm_model_init_args["seasonal_period"] = get_seasonality(data.freq)
+        return sm_model_init_args
+
+    def _fit_local_model(
+        self, timeseries: pd.Series, sm_model_init_args: dict, sm_model_fit_args: dict
+    ) -> FittedLocalModel:
+        # There is no parameter estimation involved
+        return FittedLocalModel(model_name=self.name, sm_model_init_args=sm_model_init_args, parameters=None)
+
+    def _predict_with_local_model(
+        self, timeseries: pd.Series, fitted_model: FittedLocalModel, quantile_levels: List[float]
+    ) -> pd.DataFrame:
+        seasonal_period = fitted_model.sm_model_init_args["seasonal_period"]
+        start_ts = timeseries.index.max() + pd.tseries.frequencies.to_offset(self.freq)
+        forecast_index = pd.date_range(start=start_ts, periods=self.prediction_length, freq=self.freq)
+
+        target = timeseries.values
+        forecast = {}
+        if len(target) >= seasonal_period:
+            indices = [len(target) - seasonal_period + k % seasonal_period for k in range(self.prediction_length)]
+            forecast["mean"] = target[indices].reshape(-1)
+        else:
+            forecast["mean"] = np.full(shape=[self.prediction_length], fill_value=target.mean())
+
+        for q in quantile_levels:
+            forecast[str(q)] = forecast["mean"]
```

Comment:
I see the intention here but I think this is semantically incorrect and could fail silently. Shouldn't we instead set to `NaN`?

Reply:
Using NaNs here can later cause problems in `TimeSeriesEvaluator` / `WeightedEnsemble`. If we decide to use NaNs, we will need to introduce more safeguards in those classes / logic that handles columns with NaNs as missing.

We should in general discuss how we should handle models that cannot produce probabilistic forecasts. Should we include them at all? Should we let the users decide if they are fine receiving point forecasts only? This is also very relevant for the Tabular forecasting model, we can discuss it in that PR

Reply:
does it cause problems in the sense that it raises an error? or do the NaNs propagate in a meaningful way?